File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/domain/Item.java
package com.themaker.fshmo.klassikaplus.data.domain;

public class Item {

    private String id;
    private String name; //h1
    private String description; //ключевое описание
    private String vendorCode;
    private Boolean novelty;
    private String descriptionLong;
    private String pageAlias;
    private String icon;
    private String photo;
    private String manufacturer; //производитель
    private double basePrice;
    private double discount;
    private double price;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getVendorCode() {
        return vendorCode;
    }

    public void setVendorCode(String vendorCode) {
        this.vendorCode = vendorCode;
    }

    public Boolean getNovelty() {
        return novelty;
    }

    public void setNovelty(Boolean novelty) {
        this.novelty = novelty;
    }

    public String getDescriptionLong() {
        return descriptionLong;
    }

    public void setDescriptionLong(String descriptionLong) {
        this.descriptionLong = descriptionLong;
    }

    public String getPageAlias() {
        return pageAlias;
    }

    public void setPageAlias(String pageAlias) {
        this.pageAlias = pageAlias;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public double getBasePrice() {
        return basePrice;
    }

    public void setBasePrice(double basePrice) {
        this.basePrice = basePrice;
    }

    public double getDiscount() {
        return discount;
    }

    public void setDiscount(double discount) {
        this.discount = discount;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}

File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/domain/Item.kt
package com.themaker.fshmo.klassikaplus.data.domain

import android.os.Parcel
import android.os.Parcelable
import java.io.Serializable

data class Item(var id: String?) : Serializable, Parcelable {
    override fun toString(): String {
        return "Item{" +
                "id='" + id + '\''.toString() +
                '}'.toString()
    }

    constructor(
        extId: String?,
        name: String?,
        description: String?,
        vendorCode: String?,
        novelty: Boolean,
        descriptionLong: String?,
        pageAlias: String?,
        icon: String?,
        photo: String?,
        manufacturer: String?,
        basePrice: Double,
        discount: Double,
        discountable: Boolean,
        price: Double,
        category: String?,
        categoryId: Int
    ) : this(extId)

    constructor(source: Parcel) : this(
        source.readString(),
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(id)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Item> = object : Parcelable.Creator<Item> {
            override fun createFromParcel(source: Parcel): Item = Item(source)
            override fun newArray(size: Int): Array<Item?> = arrayOfNulls(size)
        }
    }
}

